homeassistant/components/peco/sensor.py

```diff
@@ -0,0 +1,130 @@
+    SensorEntityDescription(key="customers_served", name="Customers Served"),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor platform."""
+    api = hass.data[DOMAIN][config_entry.entry_id]["api"]
+    websession = hass.data[DOMAIN][config_entry.entry_id]["websession"]
+    conf: MappingProxyType[str, Any] = config_entry.data
+
+    async def async_update_data() -> dict[str, float]:
+        """Fetch data from API."""
+        if conf["county"] == "TOTAL":
+            try:
+                data = cast(dict[str, float], await api.get_outage_totals(websession))
+            except HttpError as err:
+                raise UpdateFailed(f"Error fetching data: {err}") from err
+            except BadJSONError as err:
+                raise UpdateFailed(f"Error parsing data: {err}") from err
+            except asyncio.TimeoutError as err:
+                raise UpdateFailed(f"Timeout fetching data: {err}") from err
+            if data["percent_customers_out"] < 5:
+                percent_out = data["customers_out"] / data["customers_served"] * 100
+                data["percent_customers_out"] = percent_out
+            return data
+        try:
+            county_data = cast(
+                dict[str, float],
+                await api.get_outage_count(conf["county"], websession),
+            )
+        except HttpError as err:
+            raise UpdateFailed(f"Error fetching data: {err}") from err
+        except BadJSONError as err:
+            raise UpdateFailed(f"Error parsing data: {err}") from err
+        except asyncio.TimeoutError as err:
+            raise UpdateFailed(f"Timeout fetching data: {err}") from err
+        if county_data["percent_customers_out"] < 5:
+            percent_out = (
+                county_data["customers_out"] / county_data["customers_served"] * 100
+            )
+            county_data["percent_customers_out"] = percent_out
+        return county_data
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="PECO Outage Count",
+        update_method=async_update_data,
+        update_interval=timedelta(minutes=SCAN_INTERVAL),
+    )
+
+    await coordinator.async_config_entry_first_refresh()
+
+    county = conf["county"]
+
+    sensors = []
+    for sensor in SENSOR_LIST:
+        sensors.append(PecoSensor(hass, sensor, county, coordinator))
+    async_add_entities(
+        sensors,
+        True,
+    )
+    return
+
+
+class PecoSensor(CoordinatorEntity, SensorEntity):
+    """PECO outage counter sensor."""
+
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    _attr_icon: str = "mdi:power-plug-off"
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        description: SensorEntityDescription,
+        county: str,
+        coordinator: DataUpdateCoordinator,
+    ) -> None:
```

Comment:
Unfortunately the CoordinatorEntities generic type can't be inferred by the DataUpdateCoordinators generic type. So you could set the CoordinatorEntities data type to the same as your return type of the `update_method` so your `data` key is properly typed.
```suggestion
class PecoSensor(CoordinatorEntity[dict[str, float]], SensorEntity):
    """PECO outage counter sensor."""

    _attr_state_class = SensorStateClass.MEASUREMENT
    _attr_icon: str = "mdi:power-plug-off"

    def __init__(
        self,
        hass: HomeAssistant,
        description: SensorEntityDescription,
        county: str,
        coordinator: DataUpdateCoordinator[dict[str, float]],
    ) -> None:
```

mypy would tell you then that your `native_value` has incorrect return type in `int` since your data is `float`.